docs/src/content/hardhat-runner/docs/guides/test-contracts.md

```diff
@@ -1,3 +1,197 @@
+
+```solidity
+function withdraw() public {
+  require(block.timestamp >= unlockTime, "You can't withdraw yet");
+  require(msg.sender == owner, "You aren't the owner");
+```
+
+The first statement checks that the unlock time has been reached, and the second one checks that the address calling the contract is its owner. Let’s start by writing a test for the first pre-condition:
+
+```tsx
+it("Should revert with the right error if called too soon", async function () {
+  // ...deploy the contract as before...
+
+  await expect(lock.withdraw()).to.be.revertedWith("You can't withdraw yet");
+});
+```
+
+In the previous test we used `.to.equal`, which is part of Chai and it’s used to compare two values, but here we are using [`.to.be.revertedWith`](/hardhat-chai-matchers/docs/reference#.revertedwith). This matcher asserts that a transaction reverts, and that the reason string of the revert is equal to a given string. The `.to.be.revertedWith` matcher is not part of Chai itself; instead, it’s added by the [Hardhat Chai Matchers](/hardhat-chai-matchers) plugin, which is included in the sample project we are using.
+
+Notice that in the previous test we wrote `expect(await ...)` but now we are doing `await expect(...)`. In the first case we were comparing two values in a synchronous way; the inner await is just there to wait for the value to be retrieved. In the second case, the whole assertion is async, because it has to wait until the transaction is mined. This means that the `expect` call returns a promise that we have to wait.
+
+### Manipulating the time of the network
+
+We are deploying our `Lock` contract with an unlock time of one year. If we want to write a test that checks what happens after the unlock time has passed, we can’t wait that amount of time. We could use a shorter amount, like 5 seconds, but that’s a less realistic value and it's still a long time to wait in a test.
+
+The solution is to simulate the passage of time. This can be done with the [`time.increaseTo`](</hardhat-network-helpers/docs/reference#increaseto(timestamp)>) network helper:
+
+```tsx
+it("Should transfer the funds to the owner", async function () {
+  // ...deploy the contract...
+
+  await time.increaseTo(unlockTime);
+
+  // this will throw if the transaction reverts
+  await lock.withdraw();
+});
+```
+
+The `time.increaseTo` helper mines a new block with the given timestamp. After doing that, we can call the `withdraw` function again. In ethers, if a function reverts, the method will return a rejected promise. Awaiting the result then means that the test will fail if the function reverts.
```

Comment:
> In ethers, if a function reverts, the method will return a rejected promise. Awaiting the result then means that the test will fail if the function reverts.

This is a bit confusing.

What about something like:

As we mentioned, calling `lock.withdraw()` returns a `Promise`. If the transaction fails, the promise will be rejected. Using `await` will throw in that case, so the test will fail.

Reply:
See 1f4ede524fbd3aebc3a68c706b38b2ad89efdb65